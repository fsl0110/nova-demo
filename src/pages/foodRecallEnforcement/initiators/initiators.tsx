import React, { FC, HTMLAttributes } from "react";
import { connect } from "react-redux";
import classNames from "classnames";
import { Text, ChartHighcharts } from "../../../components";
import { AppState } from "../../../store/initialState";
import { fetchData } from "../../../store/actions/actions";
import text from "../../../utils/data/text.json";

export interface StateProps {
  /** Is loading data. */
  isLoading: boolean;
  /** Has an error. */
  hasError: boolean;
  /** Options and Data of the chart */
  chartOptions: Highcharts.Options;
}

export interface DispatchProps {
  /** Action for fetching data. **/
  fetchData: (location: string) => void;
}

export interface OwnProps {
  /** Location to fetch from Api */
  location: string;
}

export type Props = StateProps &
  DispatchProps &
  OwnProps &
  HTMLAttributes<HTMLDivElement>;

export const Initiators: FC<Props> = ({
  location,
  isLoading,
  hasError,
  chartOptions,
  fetchData,
  className
}) => (
  <div
    className={classNames(className, "flex-two-column")}
    data-test="Initiators"
  >
    <Text
      className="item"
      topicTag="h2"
      topic={text.initiators.topic}
      text={text.initiators.text}
      data-test="text"
    />
    <div className="item">
      <h3 className="horizontal-center">{text.initiators.chartTitle}</h3>
      <ChartHighcharts
        className="chart-wrapper horizontal-center"
        location="initiators"
        isLoading={isLoading}
        hasError={hasError}
        chartOptions={chartOptions}
        fetchHandler={fetchData}
        data-test="chart"
      />
    </div>
  </div>
);

export const InitiatorsConnected = connect(
  (state: AppState, ownProps: OwnProps): StateProps => ({
    isLoading: state[ownProps.location].isLoading,
    hasError: state[ownProps.location].hasError,
    chartOptions: state[ownProps.location].chartOptions
  }),
  {
    fetchData
  }
)(Initiators);
