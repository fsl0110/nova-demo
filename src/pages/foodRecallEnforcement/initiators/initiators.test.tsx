import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Initiators, Props } from "./initiators";

const testProps: AnyARecord = {
  location: "reports",
  fetchData: jest.fn(),
  chartOptions: {
    chart: {
      height: 350,
      width: 350,
      style: {
        fontFamily:
          "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen"
      }
    },
    credits: {
      enabled: false
    },
    loading: {
      hideDuration: 500,
      showDuration: 300
    },
    title: {
      text: ""
    },
    series: [{ type: "pie", data: [null] }]
  }
};

it("renders toMatchSnapshot", () => {
  expect(toJson(shallow(<Initiators {...testProps} />))).toMatchSnapshot();
});
