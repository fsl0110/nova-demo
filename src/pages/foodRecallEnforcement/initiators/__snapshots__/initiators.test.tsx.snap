// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initiator snapshots default 1`] = `
<div
  className="flex-two-column"
>
  <Text
    className="item"
    data-test="text"
    text="Manufacturers and/or distributors may initiate a recall at any time to fulfill their responsibility to protect the public health from products that present a risk of injury or gross deception, or are otherwise defective. Firms may also initiate a recall following notification of a problem by FDA or a state agency, in response to a formal request by FDA, or as ordered by FDA."
    topic="The vast majority of food recalls are voluntary"
    topicTag="h2"
  />
  <div
    className="item"
  >
    <h3
      className="horizontal-center"
    >
      Enforcement initiators
    </h3>
    <ChartHighcharts
      chartOptions={
        Object {
          "chart": Object {
            "width": 350,
          },
          "credits": Object {
            "enabled": false,
          },
          "loading": Object {
            "hideDuration": 500,
            "showDuration": 300,
          },
          "series": Array [
            Object {
              "data": Array [],
              "type": "pie",
            },
          ],
          "title": Object {
            "text": "",
          },
        }
      }
      className="chart-wrapper horizontal-center"
      data-test="chart"
      hasError={false}
      isLoading={false}
      location="initiators"
    />
  </div>
</div>
`;

exports[`Initiator snapshots is loading 1`] = `
<div
  className="flex-two-column"
>
  <Text
    className="item"
    data-test="text"
    text="Manufacturers and/or distributors may initiate a recall at any time to fulfill their responsibility to protect the public health from products that present a risk of injury or gross deception, or are otherwise defective. Firms may also initiate a recall following notification of a problem by FDA or a state agency, in response to a formal request by FDA, or as ordered by FDA."
    topic="The vast majority of food recalls are voluntary"
    topicTag="h2"
  />
  <div
    className="item"
  >
    <h3
      className="horizontal-center"
    >
      Enforcement initiators
    </h3>
    <ChartHighcharts
      chartOptions={
        Object {
          "chart": Object {
            "width": 350,
          },
          "credits": Object {
            "enabled": false,
          },
          "loading": Object {
            "hideDuration": 500,
            "showDuration": 300,
          },
          "series": Array [
            Object {
              "data": Array [],
              "type": "pie",
            },
          ],
          "title": Object {
            "text": "",
          },
        }
      }
      className="chart-wrapper horizontal-center"
      data-test="chart"
      hasError={false}
      isLoading={true}
      location="initiators"
    />
  </div>
</div>
`;
